cloud/amazon/kinesis.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = """
+---
+module: kinesis
+short_description: Creates an AWS Kinesis stream
+description:
+  - Creates an AWS Kinesis stream
+version_added: "2.0.0"
+author: "Sarah Haskins (@sarahhaskins)"
+options:
+  name:
+    description:
+      - Name of the AWS Kinesis stream to be created
+    required: true
+  shard_count:
+    description:
+      - Number of shards
+    required: true
+extends_documentation_fragment: aws
+"""
+
+try:
+    import boto3
+    import botocore
+    HAS_BOTO = True
+except ImportError:
+    HAS_BOTO = False
+
+
+class KinesisManager:
+    """Handles EC2 instance ELB registration and de-registration"""
+
+    def __init__(self, module, name=None, shard_count=None, **aws_connect_params):
+        self.module = module
+        self.name = name
+        self.shard_count = shard_count
+        self.aws_connect_params = aws_connect_params
+        self.changed = False
+        self.kinesis = self._get_connection()
+
+    def create(self):
+        """Creates the Kinesis stream if it does not already exist"""
+        if self.exists():
+            self.changed = False
+        else:
+            try:
+                self.kinesis.create_stream(StreamName=self.name, ShardCount=self.shard_count)
+                self.changed = True
+            except(botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError, StandardError), e:
+                self.module.fail_json(msg=str(e))
+
+    def exists(self):
+        """Checks if the stream by that name already exists"""
+        try:
+            self.kinesis.describe_stream(StreamName=self.name)
+            return True
+        except(botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError, StandardError), e:
+            self.module.fail_json(msg=str(e))
+
+    def _get_connection(self):
```

Comment:
Would be preferred if you used the boto3 connection function in module_utils/ec2.py

https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/ec2.py#L43